zebra-state/src/service/finalized_state.rs

```diff
@@ -378,6 +379,11 @@ impl FinalizedState {
             })
     }
 
+    /// Returns the history tree for the finalized state.
+    pub fn history_tree(&self) -> &HistoryTree {
+        todo!("add history tree to finalized state");
```

Comment:
Is this change going to be part of this PR?

If we can implement and test the non-finalized state by itself, that would make the PRs smaller. But if that's not possible, that's ok.